code/modules/crafting/craft.dm

```diff
@@ -248,6 +249,11 @@
 		else
 			for(var/i in 1 to needed_amount)
 				var/atom/movable/part_atom = locate(thing) in (surroundings - parts_used)
+				while(recipe.blacklist.Find(part_atom.type))
```

Comment:
can this be turned into a for loop instead?

Reply:
and done!